### PR TITLE
WT-8842 Create unit tests for TAILQ

### PR DESCRIPTION
**Pull request by jeremythorp**

### CODE REVIEW

File: test/unittest/tests/test_tailq.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+        testTailQWrapper.removeValue(60);
+        auto returnedItems2 = testTailQWrapper.copyItemsFromTailQ();
+        CHECK(returnedItems2 == std::list<int>{10, 20, 40, 50, 70});
+
+        // 99 isn't in the TAILQ, so this will have no effect.
+        testTailQWrapper.removeValue(99);
+        auto returnedItems3 = testTailQWrapper.copyItemsFromTailQ();
+        CHECK(returnedItems3 == std::list<int>{10, 20, 40, 50, 70});
+    }
```
Comment (korteland) [inline test/unittest/tests/test_tailq.cpp:123]: Would it be worth testing another edge case here of removing from an empty list?

**Review by korteland — COMMENTED** [review]
LGTM modulo the review comment :-)

File: test/unittest/tests/test_tailq.cpp (hunk 1 above)
Comment (jeremythorp) [reply test/unittest/tests/test_tailq.cpp:123]: Thanks and a good idea.

Comment (jeremythorp) [reply test/unittest/tests/test_tailq.cpp:123]: Now added :-)

**Review by korteland — APPROVED** [review]
LGTM, thanks for that!

**Pull request merged**